### PR TITLE
Add possibility to adjust arity level of Node in IDE

### LINKED ISSUE #1067: As a casual xoder I want adjust arity level in IDE so that variadic nodes become usable
Let’s draw the same resize handle we use in comments to allow a xoder to set arity level.

### Acceptance criteria

I create a `variadic-2` patch node, place it as a node, then

- [x] I see the resize handle
- [x] The cursor is `<->` for the whole right edge of the node
- [x] I can hold mouse button down and drag anywhere across the edge to start adjustment
- [ ] ~~The preview of the target level is shown with the blue shadow~~
- [x] An “incomplete” resize is rounded further from the current size after the cursor past ½ of a grid slot width
- [x] After mouse button up, the arity level is updated, related inputs are added/removed
- [x] If the arity level is decreased and there are incoming links to the pins that are going to disappear, the links are deleted

### PR DESCRIPTION
**Pull request by brusherru**
It closes #1067.

### CODE REVIEW

File: packages/xod-client/src/editor/components/PatchDocs.jsx [hunk 1]
```diff
@@ -172,6 +172,7 @@ const PatchDocs = ({ patch, minimal }) => {
                 position={position}
               />
             </g>
+            <rect className="PatchDocs--nodeOverlay" width="100%" height="100%" />
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/components/PatchDocs.jsx]: `Node` has `noEvents` prop

File: packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import {
+  getOffsetMatrix,
+  bindApi,
+  getMousePosition,
+} from '../modeUtils';
+
+let patchSvgRef = null;
+
+const getDesiredArityLevel = (api, mousePosition) => {
+  const deltaX = R.subtract(
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx]: Why `R.subtract`? `-` isn't good enough? 😁

File: packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    mousePosition.x,
+    api.state.dragStartPosition.x
+  );
+
+  return R.compose(
+    x => Math.max(x, 1),
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx]: `R.max(1)` ?

File: packages/xod-client/src/editor/constants.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ export const EDITOR_MODE = {
   ACCEPTING_DRAGGED_PATCH: 'accepting_dragged_patch',
   DEBUGGING: 'debugging',
   MARQUEE_SELECTING: 'marquee_selecting',
+  CHANGING_ARITY_LEVEL: 'changin_arity_level',
```
Comment (nkrkv) [inline packages/xod-client/src/editor/constants.js]: changin ← g is missing

File: packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  },
+
+  onMouseMove(api, event) {
+    const mousePosition = getMousePosition(patchSvgRef, api.props.offset, event);
+    const desiredArityLevel = getDesiredArityLevel(api, mousePosition);
+    api.setState({ mousePosition, desiredArityLevel });
+  },
+  onMouseUp(api) {
+    // TODO: Call one action!
+    api.props.actions.changeArityLevel(
```
Comment (nkrkv) [inline packages/xod-client/src/editor/containers/Patch/modes/changingArityLevel.jsx]: Obsolete TODO?

File: packages/xod-client/src/project/selectors.js [hunk 6]
```diff
@@ -208,21 +208,44 @@ const addVariadicErrors = R.curry(
   )(renderableNode)
 );
 
+/**
+ * Addds `isVariadic` flag and `arityStep` prop.
```
Comment (nkrkv) [inline packages/xod-client/src/project/selectors.js]: Adddddds

**Pull request merged**